ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -58,28 +68,36 @@ static void setupAuthHandlers(final Server server, final Router router,
     systemAuthenticationHandler.ifPresent(handler -> registerAuthHandler(router, handler));
```

Comment:
I don't think it should matter whether they're considered system authenticated before you check the path because it only has the effect of skipping any subsequent checks if they are.

Reply:
ack